presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -366,16 +367,16 @@ private void loadPartition(HivePartitionMetadata partition)
             LocatedFileStatus next = hiveFileIterator.next();
             if (isDirectory(next)) {
                 // Fail here to be on the safe side. This seems to be the same as what Hive does
-                throw new PrestoException(HIVE_PARTITION_SCHEMA_MISMATCH, "Found sub-directory in bucket directory");
+                throw new PrestoException(HIVE_INVALID_BUCKET_FILES, "Found sub-directory in bucket directory");
```

Comment:
We should include the directory name, to help the user or administrator in case it is just one bad directory of thousands:

```
throw new PrestoException(HIVE_INVALID_BUCKET_FILES, "Found sub-directory in bucket directory: " + next.getPath());